openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go

```diff
@@ -110,14 +111,25 @@ func (o *DiscoverEtcdInitialClusterOptions) Validate() error {
 }
 
 func (o *DiscoverEtcdInitialClusterOptions) Run() error {
-	_, err := os.Stat(o.MemberDir)
+
+	//Temporary hack to work with the current pod.yaml
+	var memberDir string
+	if strings.HasSuffix(o.DataDir, "member") {
+		memberDir = o.DataDir
+		o.DataDir = filepath.Dir(o.DataDir)
+	} else {
+		memberDir = filepath.Join(o.DataDir, "member")
+	}
+
+	memberDirExists := false
+	_, err := os.Stat(memberDir)
 	switch {
 	case os.IsNotExist(err):
 		// do nothing. This just means we fall through to the polling logic
 
 	case err == nil:
-		fmt.Printf(o.TargetName)
-		return nil
+		fmt.Fprintf(os.Stderr, "memberDir %s is present on %s\n", memberDir, o.TargetName)
+		memberDirExists = true
```

Comment:
looks like you've broken the "start from nothing" case.  In this case you fall through here, but the targetMember is nil a few lines down.  instead of timing out, you want to write the target name and exit cleanly.